### PR TITLE
anchor-reference-server: add reference server implementation for `GET /rate` synchronous callback API

### LINKED ISSUE #135: SEP-38: Sync Callback API: `core` interface and `anchor-reference-server` Implementation
GanttStart: 2022-03-07
GanttDue: 2022-03-29

### PR DESCRIPTION
**Pull request by marcelosalloum**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Add reference server implementation for `GET /rate` synchronous callback API.

### Why

Closes #135.

### CODE REVIEW

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/service/RateService.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    return quote;
+  }
+
+  private static class ConversionPrice {
+    private static final String fiatUSD = "iso4217:USD";
+    private static final String stellarUSDC =
+        "stellar:USDC:GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5";
+    private static final String stellarJPYC =
+        "stellar:JPYC:GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5";
+    private static final Map<Pair<String, String>, String> hardcodedPrices =
```
Comment (lijamie98) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/service/RateService.java:103]: Are we externalizing these pairs?

File: platform-apis/src/main/java/org/stellar/platform/apis/callbacks/responses/GetRateResponse.java [hunk 2]
```diff
@@ -1,18 +1,41 @@
 package org.stellar.platform.apis.callbacks.responses;
 
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.gson.annotations.SerializedName;
 import java.time.LocalDateTime;
 import lombok.Data;
 
 @Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
```
Comment (lijamie98) [inline platform-apis/src/main/java/org/stellar/platform/apis/callbacks/responses/GetRateResponse.java:10]: We are using gson as our preferred json mapper. I think this field may be ignored.
https://github.com/stellar/java-stellar-anchor-sdk/blob/main/anchor-reference-server/src/main/java/org/stellar/anchor/reference/AnchorReferenceServer.java#L20

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/controller/RateController.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  /** Gets a rate */
+  @RequestMapping(
+      value = "/rate",
+      method = {RequestMethod.GET})
+  @ResponseBody
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public GetRateResponse getRate(
+      @RequestParam() String type,
+      @RequestParam(required = false) String id,
+      @RequestParam(name = "sell_asset") String sellAsset,
```
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply anchor-reference-server/src/main/java/org/stellar/anchor/reference/controller/RateController.java]: This is the best I could get: https://github.com/stellar/java-stellar-anchor-sdk/pull/138/commits/253696e8f989ab03b9ed5d47c1cd6f2bf2b59449

[the remark this answers is not in the document]
Comment (lijamie98) [reply anchor-reference-server/src/main/java/org/stellar/anchor/reference/controller/RateController.java]: This works. If you don't mind, I can try to get JSON work later.

File: platform-apis/src/main/java/org/stellar/platform/apis/callbacks/responses/GetRateResponse.java (hunk 2 above)
Comment (lijamie98) [reply platform-apis/src/main/java/org/stellar/platform/apis/callbacks/responses/GetRateResponse.java]: That's strange.... 

We probably should review whether to stay with gson or switch to jackson later. I know Jackson is pretty old and gson is more modern I did not spend too much time researching.

**Review by lijamie98 — APPROVED** [review]
LGTM

**Pull request merged**